rest/iscn.go

```diff
@@ -0,0 +1,107 @@
+package rest
+
+import (
+	"encoding/json"
+	"log"
+	"net/url"
+
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/gin-gonic/gin"
+	iscnTypes "github.com/likecoin/likechain/x/iscn/types"
+	"github.com/likecoin/likecoin-chain-tx-indexer/db"
+)
+
+type ISCNRecordsResponse struct {
+	Records []iscnTypes.QueryResponseRecord `json:"records"`
+}
+
+func handleISCN(c *gin.Context) {
+	q := c.Request.URL.Query()
+	provided := false
```

Comment:
`provided` is confusing, probably should use something like `filtered` , `hasQuery`, `isList`